File: KHost.App/Models/SongSearchEngine.cs
namespace KHost.App.Models
{
    public class SongSearchEngine : BaseModel
    {
        public string Name { get; set; }

        public string DisplayName { get; set; }

        public bool IsLocal { get; set; } = true;

        public bool AllowDownload { get; set; } = false;
    }
}

File: KHost.App/ClientApp/src/app/models/SongSearchEngine.ts
export class SongSearchEngine {

    name: string;
    
    displayName: string;
    
    isLocal: boolean;
    
    allowDownload: boolean;

    constructor({
        name,
        displayName,
        isLocal,
        allowDownload
    }: parameters) {
        this.name = name;
        this.displayName = displayName;
        this.isLocal = isLocal ?? true;
        this.allowDownload = allowDownload ?? false;
    }
}

interface parameters {
    name: string,
    displayName: string,
    isLocal?: boolean,
    allowDownload?: boolean
}